test/encryption_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Hewlett Packard Enterprise Development LP
+
+package test
+
+import (
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/client/v1/nimbleos"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/param"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/service"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+const passphrase = "passphrase-91"
+const newPassphrase = "passphrase-92"
+
+type EncryptionWorkflowSuite struct {
+	suite.Suite
+	groupService      *service.NsGroupService
+	keyManagerService *service.KeyManagerService
+	masterKeyService  *service.MasterKeyService
+}
+
+func (suite *EncryptionWorkflowSuite) SetupSuite() {
+	groupService, err := config()
+	assert.Nilf(suite.T(), err, "Unable to connect to group, err: %v", err)
+	suite.keyManagerService = groupService.GetKeyManagerService()
+	suite.masterKeyService = groupService.GetMasterKeyService()
+	suite.deleteMasterKey()
+}
+
+func (suite *EncryptionWorkflowSuite) TearDownSuite() {
+	suite.deleteMasterKey()
+}
+
+func (suite *EncryptionWorkflowSuite) deleteMasterKey() {
+	filter := &param.GetParams{}
+	key, _ := suite.masterKeyService.GetMasterKeys(filter)
+	if len(key) != 0 {
+		err := suite.masterKeyService.DeleteMasterKey(*key[0].ID)
+		assert.Nilf(suite.T(), err, "Unable to delete key, err: %v", err)
+	}
+}
+
+func (suite *EncryptionWorkflowSuite) TestCreateMasterKey() {
+	var passphrase *string = param.NewString(passphrase)
+	key := &nimbleos.MasterKey{
+		Passphrase: passphrase,
+	}
+	_, err := suite.masterKeyService.CreateMasterKey(key)
+	assert.Nilf(suite.T(), err, "Master key creation failed, err: %v", err)
+}
+
+func (suite *EncryptionWorkflowSuite) TestCreateMasterKeyDuplicate() {
+	key := &nimbleos.MasterKey{
+		Passphrase: param.NewString("passphrase1"),
+	}
+	_, err := suite.masterKeyService.CreateMasterKey(key)
+	assert.NotNil(suite.T(), err, "Master key creation when one exists should have failed")
```

Comment:
We had a discussion earlier about state dependencies across test cases.  I don't recall the result.  Is there an intention to avoid them?

Reply:
I started combining test cases, example in protection template "TestCreateProtectionTemplate" is also updating the protection template. But if i am going overboard its looking more like one giant test case than smaller tests. I am still

Reply:
I was not sure how to achieve it besides making it one big test case. Just realized that by moving create to setup, will be able to achieve most of the test case independence. Will implement that change and push in sometime.

Reply:
Moved create in Setup.